### DISCUSSION TITLE
Tem alguma maneira de comprimir imagens usando o webpack?

### DISCUSSION THREAD

**Discussion by renedocket** [frontendbr/forum#1522] [Perguntas e Respostas]
Olá, pessoal tudo bem?

Gostaria de saber se tem alguma forma de comprimir imagens através do webpack, estou em um projeto que tem muitas imagens e isso pode acabar deixando a aplicação lenta, por esse motivo basicamente.

Podem me ajudar? Agradecido :dancer:

**Comment by wmartins** [comment]
Normalmente utilizo o [`image-webpack-loader`](https://www.npmjs.com/package/image-webpack-loader) para isso. A vantagem é que ele já agrupa diferentes otimizações para cada tipo de imagem.

Segue um exemplo:

```js
rules: [{
  test: /\.(gif|png|jpe?g|svg)$/i,
  use: [
    'file-loader',
    {
      loader: 'image-webpack-loader',
      options: {
        mozjpeg: {
          progressive: true,
          quality: 65
        },
        // optipng.enabled: false will disable optipng
        optipng: {
          enabled: false,
        },
        pngquant: {
          quality: '65-90',
          speed: 4
        },
        gifsicle: {
          interlaced: false,
        },
        // the webp option will enable WEBP
        webp: {
          quality: 75
        }
      }
    },
  ],
}]
```

**Comment by salatielsql** [comment]
> Normalmente utilizo o [`image-webpack-loader`](https://www.npmjs.com/package/image-webpack-loader) para isso. A vantagem é que ele já agrupa diferentes otimizações para cada tipo de imagem.

Parece que essa lib está com dependencias antigas, essa parace ter as mesmas features porém mais atualizada: https://www.npmjs.com/package/imagemin-webpack-plugin

**Comment by felipefialho** [comment]
Faz anos que uso o [imagemin-webpack](https://www.npmjs.com/package/imagemin-webpack)

**Comment by eliseumds** [comment]
Comprimir imagens hoje em dia é fácil e eu não deixaria mais essa responsabilidade com o Webpack (e a tonelada de dependências que esses loaders requerem). Você comprime uma vez e pronto, ao invés de deixar isso com o Webpack toda vez que estiver desenvolvendo ou fazendo deploy. Ferramentas como [optimize-images](https://github.com/victordomingos/optimize-images) e [imgp](https://github.com/jarun/imgp) podem facilmente comprimir imagens em diretórios inteiros recursivamente.

**Comment by felipefialho** [comment]
@eliseumds Bom ponto, o Netlify mesmo tem otimizações que podem ser ativadas facilmente.

**Comment by eliseumds** [comment]
@felipefialho verdade. Para imagens dinâmicas (enviadas por usuários), é o jeito.

**Comment by eliseumds** [comment]
@felipefialho opa, me confundi. Aparentemente o Netlify só comprime imagens que estiverem no repositório Git. Ele não consegue conectar com um storage externo como S3. CloudFare e Cloudinary são boas opções.
